src/docfx/lib/HttpClientUtility.cs

```diff
@@ -25,6 +26,10 @@ public static async Task<HttpResponseMessage> GetAsync(string requestUri, Config
                     // "The request message was already sent. Cannot send the same request message multiple times."
                     var message = CreateHttpRequestMessage(requestUri, config);
                     message.Method = HttpMethod.Get;
+                    if (!string.IsNullOrEmpty(etag))
+                    {
+                        message.Headers.IfNoneMatch.Add(new EntityTagHeaderValue(etag));
```

Comment:
where is the logic to handle `304 Not Modified` if the etag did match.

Reply:
https://github.com/dotnet/docfx/pull/3726/files#diff-10735dd8751bdbb809d9a5452f423fadR96